### DISCUSSION TITLE
Type Generation

### DISCUSSION THREAD

**Discussion by mledl** [ivanhofer/typesafe-i18n#641] [Q&A] [answered]
The `Locales` union type is generated based on the `/i18n/${locale}/index.ts` structure. Our application would only provide the `BaseTranslation` through `/i18n/en/index.ts`. Any other translation is loaded from an external source. Is there an easy way of generating the types here or do we have to create the corresponding file structure for each locale?

**Comment by ivanhofer** [comment, marked as answer]
You can create empty `index.js` files or override the type manually.

**Reply by mledl** [reply]
In order to keep the possibility of providing local fallbacks it would make sense to create the `index.js` files. Or would you go for overriding the `Locales` type?

**Reply by ivanhofer** [reply]
Depends on your setup..
You need to implement a custom locale loading strategy. You can also just wrap the `typesafe-i18n` functionality or just override the genereated functions and set the [`generateOnlyTypes`-option](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/generator#generateonlytypes) to `true`.
